### PR TITLE
Migrate to SLE BCI images

### PR DESCRIPTION
**Pull request by thomasferrandiz**
This PR removes the usage of alpine images and replaces them with SLE BCI images.

### CODE REVIEW

**Comment by pgonin** [comment]
poke @macedogm

**Comment by pgonin** [comment]
build fail because of `no matching manifest for linux/arm/v7 in the manifest list entries`
I don't think we need to keep building for arm v7

**Comment by macedogm** [comment]
@pgonin I spoke with Caleb about `armv7` and we don't need to build for this arch, so it's safe to remove it from the build pipeline. I had do to it also for some of the PRs that I opened. We need to build and ship only `amd64`, `s390x` and `arm64`.

**Review by macedogm — CHANGES_REQUESTED** [review]
We just need to remove the build step for `armv7` from Drone, then LGTM.

**Comment by dweomer** [comment]
> @pgonin I spoke with Caleb about `armv7` and we don't need to build for this arch, so it's safe to remove it from the build pipeline. I had do to it also for some of the PRs that I opened. We need to build and ship only `amd64`, `s390x` and `arm64`.

This is incorrect: SUC is leveraged to upgrade k3s which still ships for arm v7.

File: e2e/cluster/local/images/k3s/Dockerfile [hunk 1]
```diff
@@ -1,12 +1,11 @@
 # Install the all-in-one binary so we can copy our run-time images into the image
 # which helps avoid pulling them when running e2e tests.
-ARG ALPINE="library/alpine:3.14"
-FROM ${ALPINE} AS k3s
+ARG SLES="registry.suse.com/suse/sle15:15.3"
```
Comment (dweomer) [inline e2e/cluster/local/images/k3s/Dockerfile:3]: Why SLES here instead of BCI? Unpacked on my system, the SLES image is 7 times larger than Alpine.

**Comment by dweomer** [comment]
This is also a go version bump. In trying to see if the go.mod/go.sum would change with a `go mod tidy` I ran into:
```
$ go mod tidy
go: downloading github.com/docker/distribution v2.8.0+incompatible
verifying github.com/docker/distribution@v2.8.0+incompatible: checksum mismatch
        downloaded: h1:u9vuu6qqG7nN9a735Noed0ahoUm30iipVRlhgh72N0M=
        go.sum:     h1:l9EaZDICImO1ngI+uTifW+ZYvvz7fKISBAKpg+MbWbY=

SECURITY ERROR
This download does NOT match an earlier download recorded in go.sum.
The bits may have been replaced on the origin server, or an attacker may
have intercepted the download attempt.

For more information, see 'go help module-auth'.
```

Fixed via https://github.com/rancher/system-upgrade-controller/pull/196

File: e2e/cluster/local/images/k3s/Dockerfile (hunk 1 above)
Comment (thomasferrandiz) [reply e2e/cluster/local/images/k3s/Dockerfile:3]: In fact it looks like `bci-base` and `sle15` images are identical.
If you check here, they have the same digest:
https://registry.suse.com/static/suse/sle15sp3/index.html
https://registry.suse.com/static/bci/bci-base/index.html

But I agree it is unfortunate that they are so large.

There are also `bci-minimal` and `bci-micro` images which are small but they are still in Tech Preview.
As I understand, @macedogm is preparing a proposal to use them with sidecars but I don't know if we can use that yet.

Comment (macedogm) [reply e2e/cluster/local/images/k3s/Dockerfile:3]: I'm discussing internally about which image to "standardize". I don't believe that we will follow with the idea of the sidecar, but actually with a mix of `bci-micro` and `bci-base/sle15` (when really needed). Let's see how it'll progress.

Comment (cbron) [reply e2e/cluster/local/images/k3s/Dockerfile:3]: `registry.suse.com/suse/sle15:15.3` is sle-bci, not full SLES.

**Pull request merged**